uint8 l_data
